src/concrete/ml/quantization/post_training.py

```diff
@@ -23,6 +23,55 @@
 from .quantized_ops import QuantizedBrevitasQuant
 from .quantizers import QuantizationOptions, QuantizedArray, UniformQuantizer
 
+# pylint: disable=too-many-lines
+
+
+def get_n_bits_dict_trees(n_bits: Union[int, Dict[str, int]]) -> Dict[str, int]:
+    """Convert the n_bits parameter into a proper dictionary for tree based-models.
+
+    Args:
+        n_bits (int, Dict[str, int]): number of bits for quantization, can be a single value or
+            a dictionary with the following keys :
+            - "op_inputs" (mandatory)
+            - "op_leaves" (optional)
+            When using a single integer for n_bits, its value is assigned to "op_inputs" and
+            "op_leaves" bits.
+
+    Returns:
+        n_bits_dict (Dict[str, int]): A dictionary properly representing the number of bits to use
+            for quantization.
+    """
+
+    assert_true(
+        isinstance(n_bits, int)
+        or (isinstance(n_bits, Dict) and set(n_bits.keys()).issubset({"op_inputs", "op_leaves"})),
+        "Invalid n_bits, either pass an integer or a dictionary containing integer values for "
+        "the following keys:\n"
+        "- `op_inputs` and `op_leaves` (mandatory)",
+    )
+
+    n_bits_dict: Dict = {}
+
+    # If a single integer is passed, we use a default value for the model's input and
+    # output bits
+    if isinstance(n_bits, int):
+        n_bits_dict = {
+            "op_inputs": n_bits,
+            "op_leaves": n_bits,
```

Comment:
now that I think of it, I definitively get the relevancy behind having an op_leaves instead of op_weights. However, we are therefore introducing a new name and making this n_bits parameter more complex to users (which are already not supposed to be familiar with it) , so maybe we should discuss about that.

Personally, I think I wouldn't be against keeping the op_weights even for trees for that particular reason

what do you think @zama-ai/machine-learning ?

Reply:
In our first discussion with @andrei-stoian-zama on this subject, we thought about quantifying op_leaves only when TREES_USE_FHE_SUM is enabled. 

Today, it is disabled by default and we intend to enabled it when all the necessary security tools are ready. 

Then, after consideration, I think it's a great feature to add a dict for n_bits for tree. 
Even by reducing the precision of the leaves, the accuracy is preserved for the classifiers, a little less for the regressors, and on top of that, having it allows us to go further with the trees (more trees, more features, more examples than before). 

so if we validate the integration of the dict for trees, we'll have to agree on the name op_leaves, op_output, or op_weights. 
and we'll have to explain it the doc:

Reply:
yes I was only talking about adding this `op_leaves` naming through `get_n_bits_dict_trees` instead of keeping things simple as they already are with `get_n_bits_dict` (like linear models, which don't use `model_*` attributes)